flow/Promise.js

```diff
@@ -45,3 +45,7 @@ declare class Promise<+R> {
 
   static cast<T>(object?: T): Promise<T>;
 }
+
+declare module 'Promise' {
```

Comment:
This is only needed because AccessibilityInfo is requiring the Promise module instead of using Promise from global. That seems weird. Does everything else use it from global?

Reply:
It also fixes this Flow error that is showing up for me locally and some other people:

```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ Libraries/Core/polyfillPromise.js:21:33

Importing from an untyped module makes it any and is not safe! Did you mean to add // @flow to the top of Promise?
(untyped-import)

     18│  * If you don't need these polyfills, don't use InitializeCore; just directly
     19│  * require the modules you need from InitializeCore for setup.
     20│  */
     21│ polyfillGlobal('Promise', () => require('Promise'));
     22│

```

I don't know why this error isn't showing up on CircleCI (I also did a clean re-install of all `node_modules` and this still appears)

Reply:
This fix seems detrimental to that particular issue. `require('Promise')` should be resolving to `Libraries/Promise.js` via Haste (which does have @flow at the top so there is something else weird happening). By specifying `declare module 'Promise' {` in `flow/Promise.js` it is bypassing the haste module which is probably not what we want.

Can you try running flow get-def on `require('Promise')` in the statement you have above and see where Flow is trying to resolve that module from and debug from there?

Reply:
It's resolving to `node_modules/promise/index.js`.

The particular version of `promise` RN is using doesn't have a Flow type stub which is probably why this is showing up as `any`. The latest version (`8.0.2`) has it, however:

https://github.com/then/promise/commit/c98808978c0eda314c36268e284da219064348cb

A less ideal but simpler alternative however could be to just remove `strict-local` on `polyfillPromise`, since it's (supposed to be) the only place where promise is required directly.

(I am still not sure why it's working on CircleCI but not on my machine however 🤔 )